---
title: "Analysis Report for Combination Module"
author: 'IDEA: Interactive Differential Expression Analyzer'
date: '`r format(Sys.time(), format = "%H:%M:%S, %B %d %Y")`'
output:
  word_document: default
html_document:
  mathjax: http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML
---
```{r setup, echo=FALSE}
knitr::opts_chunk$set( fig.width = 9, fig.height = 9, dpi = 72)
```
This is an R Markdown document for Combination Analysis of [IDEA: Interactive Differential Expression Analyzer](http://idea.biocuckoo.org). The following plots are plotted in R [[1](#ref1)] with ggplot2[[2](#ref2)] (for histogram) and VennDiagram[[3](#ref3)] (for Venn diagram)). This analysis report is presented in HTML via rmarkdown [[4](#ref4)].   

Citation: This work is in process of publishing, citation method will be post here as soon as possible. Check out the IDEA website above.     

```{r global_options, include=FALSE, echo=FALSE, warning=FALSE}
#setwd(tempdir())
load("combinationAnalysis.RData")
#p1 basic information  
      #exprimental design plist[[1]][1]
      #select paird plist[[1]][[2]]
      #SAMseq resample number plist[[1]][3]
      #fdr cutoff plist[[1]][4]
    # #heatmap genenumber plist[[1]][5]
library(ggplot2)
library(gplots)
library(RColorBrewer)
library(scales)
library(pheatmap)
library(plyr)
library(labeling)
library(stringr)
library(rmarkdown)
#library(S4Vectors)
library(samr)
library(stringr)
```

#Introduction
Count data, as generated by various high-throughput sequencing methods such as RNA-Seq[[5](#ref5), [6](#ref6)], Tag-Seq[[7](#ref7), [8](#ref8)], and ChIP-Seq[[9](#ref9)], has been more and more used to represent the abundance of genes/features at RNA/DNA level since read count and abundance are linearly related [[6](#ref6)]. Also in RNA-Seq, variation caused by replicate is low, which makes RNA-Seq count data advantageous for differential expressed gene discovery[[10](#ref10)]. Differential expression (DE) analysis typically works with following questions: choice of normalization and noise control method[[6](#ref6), [8](#ref8)]; choice of data distribution given numbers of replicates[[8](#ref8)]; choice of assessment of statistical significance of DE detection[[11](#ref11)].   

#Basic Information
##Experiment Design
In combination analysis, experiment design is set as standard procedure of package-specific analysis, which is defined in "New" module before analysis starts. Note that some packages are not applicable for a certain experiment type (Table 1). in Multi-factor Design, factor of interest is defined as first column of counts table.   

In this case, experiment type was stated as **`r as.character(plist[[1]][1])`**. And condition **`r as.character(plist[[1]][[2]])[1]`** and condition **`r as.character(plist[[1]][[2]])[2]`** were selected for differential expression analysis.
   
```{r, echo=FALSE}
htmltools::HTML('  
<div align="center">
Table 1: Package availability for different experiment design types<br/>
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
            <td style="border-width: medium thin medium 0">&nbsp;Package</td>
            <td style="border-width: medium thin medium 0">&nbsp;Standard Comparison</td>
            <td style="border-width: medium thin medium 0">&nbsp;Multi-factor Design</td>
            <td style="border-width: medium thin medium 0">&nbsp;Without Replicates</td>
        </tr>
         <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;DESeq</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;edgeR</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;NOISeq</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Not applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;PoissonSeq</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Not applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Not applicable</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin medium 0">&nbsp;SAMseq (samr)</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Not applicable</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Not applicable</td>
        </tr>
</table>
</div>   
')
```
##Normalization Method
Since packages typically have different choices of normalization methods, to combine the results across packages, counts data inputted by user is first normalized using method chosen in Data Exploration (Table 2) before being analyzed by packages respectively.   
In this case, normalization method is **`r as.character(plist[[1]][[3]])`**.  

<div align="center">
Table 2: Normalization methods in Data Exploration<br/>
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
          <td style="border-width: medium thin medium 0">&nbsp;Method</td>
          <td style="border-width: medium thin medium 0">&nbsp;Abbreviation</td>
          <td style="border-width: medium thin medium 0">&nbsp;Summary</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Reads per Kilobase per Million Reads (default)</td>
          <td style="border-width: 0 thin thin 0">&nbsp;RPKM</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Counts per kilobase per million mapped reads or total number of reads in library calculated</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Trimmed Mean of M values</td>
          <td style="border-width: 0 thin thin 0">&nbsp;TMM</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Weight taken from delta method on binomial data, then trimmed weighted means calculated</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Upper Quartile</td>
          <td style="border-width: 0 thin thin 0">&nbsp;UQ</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Features that are zero in all library removed, scale factor calculated from a upper quartile of counts for each library</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin medium 0">&nbsp;None</td>
          <td style="border-width: 0 thin medium 0">&nbsp;None</td>
          <td style="border-width: 0 thin medium 0">&nbsp;All scaling factors set to 1</td>
        </tr>
</table>
</div>

##Advanced Option: Selected Packages  

In IDEA, five R/Bioconductor packages are available for combination analysis of dfferential expression (DE) features, including DESeq2, edgeR, NOISeq, PoissonSeq and SAMseq.   
Users can select their favorite combination of different packages to perform the combination analysis.  
In this case, the following packages were selected for combination analysis: **`r as.character(plist[[1]][4])`**.

##Advanced Option: Rank Aggregation Method    
Several rank lists are generated by different packages after analysis. To provide an integrated view of differentially expressed features, a Score of rank aggregation result is provided. Rank Aggregation Method, that is, calculation method of Score can be selected and meanings of parameters are listed below.    
<div align="center">
Table 3: Rank Aggregation Methods in Combination Analysis<br/>
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
          <td style="border-width: medium thin medium 0">&nbsp;Parameter of Rank Aggregation Method</td>
          <td style="border-width: medium thin medium 0">&nbsp;Meaning</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;RRA (default)</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Defined as result of Robust Rank Aggregation [<a href="#ref14">14</a>]</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;min</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Defined as minimum value of ranks</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;geom.mean</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Defined as geometric mean value of ranks</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;mean</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Defined as mean value of ranks</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;median</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Defined as median value of ranks</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin medium 0">&nbsp;Stuart</td>
          <td style="border-width: 0 thin medium 0">&nbsp;Defined as means of a variety of statistical tests, as in Ref [<a href="#ref15">15</a>]</td>
        </tr>
</table>
</div>

#Analysis Result
##Differentially Expressed Features Identified by Packages    
A histogram is adopted to visualize total number of differentially expressed features identified by each package.    

<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
print(plist[[2]])
```
<br/>
Figure 1: Histogram of differential expression total numbers of features features by each package
</div>

##Venn of DE Features Identified by Packages

The Venn diagram visualizes the overlapping differential expression (DE) features identified by each package. In the diagram, each oval represents a set of detected DE features in a certain package, illustrated by notes beside the oval. And the overlapping areas shown in the diagram represent all possible logical relations of DE features identification by differential packages. Digits are plotted in such areas, indicating number of DE features that follow corresponding logical relation. Total numbers of packages can be interactively changed as demonstrated in Basic Introduction of combination analysis above.    

<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
grid.draw(plist[[3]])
```
<br/>
Figure 2: Venn diagram of differential expression features analysis in each selected packages
</div>

##Feature Weight Table   

Identification details of each feature is shown in the feature weight table with interactive sorting options built-in at headers. A feature weight table in .csv format is available for download from web page.   
To better evaluate levels of DE features detected by different packages, a Recommendation Value (R-value) is introduced. To integrate these rank list, the method of robust rank aggregation (RRA) is applied [[14](#ref14)]. A Score is presented as results of rank list integration via corresponding R package `RobustRankAggreg`.  
Identification status of features in each selected packages and statistic values are listed in columns. The interpretation of headers is shown in the Table 3. 

<div align="center">
Table 4: Interpretation of headers in feature weight table<br/>
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
            <td style="border-width: medium thin medium 0">&nbsp;Headers</td>
            <td style="border-width: medium thin medium 0">&nbsp;Interpretation</td>
        </tr>
         <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;FeatureID</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Feature identifier</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;Package Name</td>
            <td style="border-width: 0 thin thin 0">&nbsp;(DESeq, edgeR, NOISeq, PoissonSeq, and/or SAMseq)<br/>Identification status of features in corresponding package; in .csv file, identified features are tagged by "Positive" and unidentified features by "Negative"; <br/>on web pages, it is tagged by "Check" and "Cross" marks</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;Mean</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Mean of feature expression</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;LogFC</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Logarithm (base 2) of the fold change</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;Rankmean</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Mean rank among the selected packages for combination analysis</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin medium 0">&nbsp;Score</td>
            <td style="border-width: 0 thin medium 0">&nbsp;Intergration score of rank lists of DE features by robust rank aggregation (RRA)</td>
        </tr>
</table>
</div>   
   
#References

##### <a name="ref1"></a>1.  R Core Team, *R: A language and environment for statistical computing*, 2014, R Foundation for Statistical Computing: Vienna, Austria.
##### <a name="ref2"></a>2.	Wickham, H., *ggplot2: elegant graphics for data analysis*, 2009, Springer New York.
##### <a name="ref3"></a>3.	Chen, H., *VennDiagram: Generate high-resolution Venn and Euler plots*, 2014.
##### <a name="ref4"></a>4.	JJ Allaire, J.M., Yihui Xie, Hadley Wickham, Joe Cheng and Jeff Allen, *rmarkdown: Dynamic Documents for R*, 2014.
##### <a name="ref5"></a>5.	Nagalakshmi, U., et al., *The transcriptional landscape of the yeast genome defined by RNA sequencing.* Science, 2008. **320**(5881): p. 1344-9.
##### <a name="ref6"></a>6.	Mortazavi, A., et al., *Mapping and quantifying mammalian transcriptomes by RNA-Seq.* Nat Methods, 2008. **5**(7): p. 621-8.
##### <a name="ref7"></a>7.	Morrissy, A.S., et al., *Next-generation tag sequencing for cancer gene expression profiling.* Genome Res, 2009. **19**(10): p. 1825-35.
##### <a name="ref8"></a>8.	Anders, S. and W. Huber, *Differential expression analysis for sequence count data.* Genome biol, 2010. **11**(10): p. R106.
##### <a name="ref9"></a>9.	Robertson, G., et al., *Genome-wide profiles of STAT1 DNA association using chromatin immunoprecipitation and massively parallel sequencing.* Nat Methods, 2007. **4**(8): p. 651-7.
##### <a name="ref10"></a>10.	Marioni, J.C., et al., *RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays.* Genome Res, 2008. **18**(9): p. 1509-17.
##### <a name="ref11"></a>11.	Robinson, M.D., D.J. McCarthy, and G.K. Smyth, *edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.* Bioinformatics, 2010. **26**(1): p. 139-40.
##### <a name="ref12"></a>12.	Robinson, M.D. and A. Oshlack, *A scaling normalization method for differential expression analysis of RNA-seq data.* Genome Biol, 2010. **11**(3): p. R25.
##### <a name="ref13"></a>13.	Bullard, J.H., et al., *Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments.* BMC Bioinformatics, 2010. **11**: p. 94.
##### <a name="ref14"></a>14.  Kolde, R., et al., *Robust rank aggregation for gene list integration and meta-analysis.* Bioinformatics, 2012. **28**(4): p. 573-80.
##### <a name="ref15"></a>15.  Stuart, J.M., et al., *A gene-coexpression network for global discovery of conserved genetic modules.* Science, 2003. **302**(5643): p. 249-55.     
