---
title: "Differential Analysis Report for edgeR "
author: "IDEA: Interactive Differential Expression Analyzer"
date: '`r format(Sys.time(), format = "%H:%M:%S, %B %d %Y")`'
output:
  word_document: default
  html_document:
    mathjax: http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML
---
```{r setup, echo=FALSE}
knitr::opts_chunk$set( fig.width = 9, fig.height = 9, dpi = 72)
```
This is an R Markdown document for edgeR analysis of [IDEA: Interactive Differential Expression Analyzer](http://idea.biocuckoo.org). Plots in Analysis module (plotted in R [[1](#ref1)] with ggplot2[[2](#ref2)] (all plots except heat maps) and pheatmap[[3](#ref3)]) are presented in HTML file via rmarkdown [[4](#ref4)]. For figures of higher resolution, please download from website directly.    
Citation: This work is in process of publishing, citation method will be post here as soon as possible. Check out the IDEA website above.    

```{r global_options, echo=FALSE, warning=FALSE}
#setwd(tempdir())
load("edgeRAnalysis.RData")
#p1 basic information 
#       plist[[1]][[1]] #exprimental design
#       plist[[1]][[2]]#select paird
#       plist[[1]][[3]]#normalization method
#       plist[[1]][[4]]#filter by
#       plist[[1]][[5]]  #heatmap topgene
# plist[[1]][[6]]#normalized factor (data.frame)

# plist[[1]][[7]]      
#if multi factor
        #intererst factor
      #if withou replicate  
        #bcv
#p2 getCdsData
#p3 getDEseqMAplot()
#p4 DEseqVolcanoPlot()
#p5 DESeqHeatmapPlotfunction()
#p6 p-value distribution
library(ggplot2)
library(gplots)
library(RColorBrewer)
library(scales)
library(pheatmap)
library(plyr)
library(labeling)
library(stringr)
library(edgeR)
library(rmarkdown)
#library(S4Vectors)
library(stringr)
library(knitr)
```

#Introduction   
Count data, as generated by various high-throughput sequencing methods such as RNA-Seq[[5](#ref5), [6](#ref6)], Tag-Seq[[7](#ref7), [8](#ref8)], and ChIP-Seq[[9](#ref9)], been more and more used to represent the abundance of genes/features at RNA/DNA level since read count and abundance are linearly related [[6](#ref6)]. Also in RNA-Seq, variation caused by replicate is low, which makes RNA-Seq count data advantageous for differential expressed gene discovery[[10](#ref10)]. DE analysis typically works with following questions: choice of normalization and noise control method [[6](#ref6), [8](#ref8)]; choice of data distribution given numbers of replicates[[8](#ref8)]; choice of assessment of statistical significance of DE detection[[11](#ref11)].   
   
edgeR[[11](#ref11)] is an R/Bioconductor package for DE analysis based on count data. It adopts negative binomial model, which is commonly used to handle overdispersion, for data distribution. The updated version edgeR-robust[[12](#ref12)], published in 2014, has refined negative binomial model in generalized linear model (GLM). Several options for normalization are available in edgeR, including the Relative Log Expression (RLE, as default) [[11](#ref11)], the Trimmed Mean of M values (TMM) [[13](#ref13)], the Upper Quartile (UQ) [[14](#ref14)] and none. Fisher’s exact test is adopted for DE test, and Benjamini and Hochberg method for multiple testing.   
   
In IDEA, edgeR(version 3.8.3) is employed. For more information on edgeR, please refer to their publications (References [[11](#ref11)] and [[12](#ref12)] ) and package manual.   

#Basic Information   

##Experimental Design   

In IDEA, a raw count table should be uploaded and experimental design be clarified. Optionally, experimental design can be one of Standard Comparison, Multi-factors Design and Without Replicates (not recommended). Then a pair of conditions should be selected to carry out differential expression analysis.   
Specifically, edgeR is applicable for all three experiment types. Note that in Multi-factor Design, factor of interest is defined as first column of counts table,   
In these case, experimental design was stated as **`r plist[[1]][1]`**. And conditions **`r as.character(plist[[1]][[2]])[1]`** and **`r as.character(plist[[1]][[2]])[2]`** were selected for differential expression analysis.   

##Advanced Options   
###Normalization Methods   
In edgeR analysis, several advanced options are provided, including normalization methods, filter number, and false discovery rate threshold. Four normalization methods are available in package edgeR, as introduced in Introduction above, whose details are listed in Table 1.   
<div align="center">
Table 1 Normalization methods in edgeR
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
          <td style="border-width: medium thin medium 0">&nbsp;Method</td>
          <td style="border-width: medium thin medium 0">&nbsp;Abbreviation</td>
          <td style="border-width: medium thin medium 0">&nbsp;Summary</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Relative Log Expression(default)[<a href="#ref11">11</a>]</td>
          <td style="border-width: 0 thin thin 0">&nbsp;RLE</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Median library from geometric mean of all columns calculated, the median ratio of each column to the median library taken as scale factor</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Trimmed Mean of M values[<a href="#ref11">11</a>]</td>
          <td style="border-width: 0 thin thin 0">&nbsp;TMM</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Weight taken from delta method on binomial data, then trimmed weighted means calculated</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin thin 0">&nbsp;The Upper Quartile[<a href="#ref14">14</a>]</td>
          <td style="border-width: 0 thin thin 0">&nbsp;UQ</td>
          <td style="border-width: 0 thin thin 0">&nbsp;Features that are zero in all library removed, scale factor calculated from a upper quartile of counts for each library</td>
        </tr>
        <tr>
          <td style="border-width: 0 thin medium 0">&nbsp;None</td>
          <td style="border-width: 0 thin medium 0">&nbsp;None</td>
          <td style="border-width: 0 thin medium 0">&nbsp;ll normalization scaling factor set to 1</td>
        </tr>
</table>
</div>
   
In this case, normalization method was set as **`r plist[[1]][[3]]`**.   

###Dispersion Estimating Method   
In edgeR, two methods for dispersion estimation is provided. Cox-Reid profile-adjusted likelihood, combined with generalized linear model (GLM) is available for all experiment types. And quantile-adjusted conditional maximum likelihood (qCML) method is only applicable for experiments with a single factor.    
In this case, **`r plist[[1]][[7]]`** is employed for dispersion estimating.    

###Filter of Low Abundance Features   
Filtering out low abundance tags/features can help improve performance of DE discovery. After selecting filter numbers in the drop-down menu titled "Filter your dataset by", only tags/features with total sums of counts above the selected cutoff are used.   

In this case, filter number was set as **`r plist[[1]][[4]]`**.   

#Analysis Result   
##Differential Expression Table   
After analysis, a table containing information of all differentially expressed genes is presented with interactive options. Implication of nouns in header is explained in Table 1.
Note that in different packages, same noun in header can have different implication. For example, *p*-values in DESeq are obtained by Wald test, but in edgeR *p*-values are obtained by Fisher's exact test.   
   
<div align="center">
Table 2: Interpretation of headers of differential expression table in edgeR
<table cellpadding="5" cellspacing="0" border="1" frame=hsides rules=all style="border-color: #000000">
        <tr>
            <td style="border-width: medium thin medium 0">&nbsp;Headers</td>
            <td style="border-width: medium thin medium 0">&nbsp;Interpretation</td>
        </tr>
         <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;FeatureID</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Feature identifier</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;logFC</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Logarithm (base 2) of the fold change, fold change is defined as counts of Condition2 divided by counts of Condition1</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;logCPM</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Average log2-counts-per-million</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin thin 0">&nbsp;PValue</td>
            <td style="border-width: 0 thin thin 0">&nbsp;Two sided p-value of Fisher’s exact test</td>
        </tr>
        <tr>
            <td style="border-width: 0 thin medium 0">&nbsp;FDR</td>
            <td style="border-width: 0 thin medium 0">&nbsp;False discovery rate</td>
        </tr>
</table>
</div>
   
##Variance Estimation

The variance estimation plot is used to visualize the tag-wise negative binomial dispersions, thus illustrating relationship of dispersion caused and abundance tags. Specifically in edgeR, the dispersion is calculated by function `estimateCommonDisp()` and `estimateTagwiseDisp()` successively, and plotted by function `plotBCV()`, all parameters are set as default. For more details, please refer to edgeR manual.    
The tag/feature-wise estimation are shown as black dots, with average log2-transformed counts-per-million (CPM) as x-axis, and biological coefficient of variation (BCV) as y-axis. The common dispersion, which is estimated across all tags/features, is plotted as a red line parallel to x-axis
   
<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
plotBCV(plist[[2]])
```
<br/>
Figure 1: Variance estimation plot for edgeR
</div>

##MA-Plot

A MA-Plot can give a quick overview of the distribution of data. The log2–transformed fold change is plotted on the y-axis and the average count (normalized by size factors) is on the x-axis. The genes with adjusted *p*-value less than the FDR are colored red, while other genes are colored black.    
<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
print(plist[[3]])
```
<br/>
Figure 2: MA-plot for edgeR
</div>

##Normalized Size Factors
Different samples may have different sequencing depth. Thus it is necessary to estimate the relative size factors of each sample, and divide the samples by the size factors separately.    
Table of normalized size factors shows the normalized size factors of each sample. In the header, *Group* represents conditions, *lib.size* represents size of the library, *norm.factors* is the normalized size factors.  
<div align="center">
Table 2: Normalized size factors calculated when normalizing in edgeR   
```{r results="hide", echo=FALSE, warning=FALSE}
#plist[[1]][[6]]
kable(plist[[1]][[6]], digits = 4, align=c("c"), format="markdown")
```
</div>

##Volcano Plot
An overview of the number of differential expression genes can be shown in the volcano plot. The log2-transformed fold change is on the x-axis, the y-axis represents the $–log_{2}(\mathit{p}value)$. The threshold of *p*-value is `r plist[[4]][[2]] `, and fold change threshold is `r plist[[4]][[3]]`. Highly differential expressed genes are colored blue, while others are in red.   
<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
#volvanolist[[1]] print p
# volvanolist[[2]] #VolcanoPlot pcut off
# volvanolist[[3]] #VolcanoPlot FC cutoff
volvanolist<-plist[[4]]
print(volvanolist[[1]])
```
<br/>
Figure 3  Volcano plot of DE features detected by edgeR
</div>

##Heat Map of Differential Expressed Genes   

Heat map can graphically display the differential expression table, and each square (pixel) represents the value of a feature in a sample and colored accordingly. Here, heat map of differential expressed features is plotted via R package *pheatmap*. Features are arranged in columns (samples) and rows (features) as in the original data matrix. Up-regulated differential expression features are colored red in heat map, while the down-regulated colored green. Hierarchical clustering results of features and samples are shown in dendrogram on the left and upper side of heat map, respectively.    
Numbers of features to display as rows, the appearance of dendrogram on both left and upper side, and the appearance of color key are all interactively changeable. The data scaling of heat map can be one of "none", "row", and "column", as chosen by user. The color is scaled by $log_{10}(Normalized Reads Count + 1)$.    
In this case, data is centered and scaled in the **`r as.character(plist[[5]][[3]])`** direction. For more information on parameter settings, please refer to the manual of package *pheatmap* (as in References [[3](#ref3)]).   
  
<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
#heatmapO[[1]]  data
#heatmapO[[2]] a logical vector with two elements 
#heatmapO[[3]] scalling method
#heatmapO[[3]] legend options
heatmapO=plist[[5]]
pheatmap(heatmapO[[1]], color=greenred(75),border_color=NA,cluster_rows=heatmapO[[2]][1],cluster_cols= heatmapO[[2]][2],scale=heatmapO[[3]],legend=heatmapO[[4]])
```
<br/>
Figure 4 Heat map of differential expressed genes, top `r plist[[1]][5]` DE features shown with lowest false discover rate (FDR) value
</div>

##FDR Distribution Plot
False discover rate (FDR) distribution plot visualizes distribution of FDR in DE test. In edgeR, Fisher’s exact test is adopted for differential expression test, and Benjamini-Hochberg procedure for multiple testing. FDR distribution plot uses FDR as x-axis and percentage of different groups of x value as y-axis, and colors significant and not significant groups differently.   

<div align="center">
```{r results="hide", echo=FALSE, warning=FALSE}
print(plist[[6]])
```
<br/>
Figure 5  FDR distribution plot in edgeR
</div>
#References
##### <a name="ref1"></a>1. R Core Team, *R: A language and environment for statistical computing*, 2014, R Foundation for Statistical Computing: Vienna, Austria.
##### <a name="ref2"></a>2. Wickham, H., *ggplot2: elegant graphics for data analysis*, 2009, Springer New York.
##### <a name="ref3"></a>3. Kolde,R., *pheatmap: Pretty Heatmaps*, 2013.
##### <a name="ref4"></a>4. JJ Allaire, J.M., Yihui Xie, Hadley Wickham, Joe Cheng and Jeff Allen, *rmarkdown: Dynamic Documents for R*, 2014.
##### <a name="ref5"></a>5. Nagalakshmi, U., et al., *The transcriptional landscape of the yeast genome defined by RNA sequencing.* Science, 2008.**320**(5881): p. 1344-9.
##### <a name="ref6"></a>6. Mortazavi, A., et al., *Mapping and quantifying mammalian transcriptomes by RNA-Seq.* Nature methods, 2008.**5**(7): p. 621-628.
##### <a name="ref7"></a>7. Morrissy, A.S., et al., *Next-generation tag sequencing for cancer gene expression profiling.* Genome Res, 2009.**19**(10): p. 1825-35.
##### <a name="ref8"></a>8. Anders, S. and W. Huber, *Differential expression analysis for sequence count data.* Genome biol, 2010.**11**(10): p. R106.
##### <a name="ref9"></a>9. Robertson, G., et al., *Genome-wide profiles of STAT1 DNA association using chromatin immunoprecipitation and massively parallel sequencing.* Nat Methods, 2007.**4**(8): p. 651-7.
##### <a name="ref10"></a>10. Marioni, J.C., et al.,*RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays*. Genome Res, 2008.**18**(9): p. 1509-17.
##### <a name="ref11"></a>11. Robinson, M.D., D.J. McCarthy, and G.K. Smyth,*edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.* Bioinformatics, 2010.**26**(1): p. 139-140.
##### <a name="ref12"></a>12. Zhou, X., H. Lindsay, and M.D. Robinson, *Robustly detecting differential expression in RNA sequencing data using observation weights.* Nucleic acids research, 2014: p. gku310.
##### <a name="ref13"></a>13.  Robinson, M.D. and A. Oshlack, *A scaling normalization method for differential expression analysis of RNA-seq data.* Genome Biol, 2010. **11**(3): p. R25.
##### <a name="ref14"></a>14.	Bullard, J.H., et al., *Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments.* BMC Bioinformatics, 2010. **11**1: p. 94.